File: update_db.py
import mysql.connector

# Database connection parameters
db_config = {
    "host": "localhost",
    "user": "root",
    "password": "",
    "database": "sysarch"
}

try:
    # Create connection
    conn = mysql.connector.connect(**db_config)
    cursor = conn.cursor()
    
    print("Connected to database successfully.")
    
    # SQL query to alter the computer_status table
    sql = """
    ALTER TABLE computer_status 
    MODIFY COLUMN status ENUM('vacant', 'occupied', 'in_class', 'maintenance') NOT NULL
    """
    
    # Execute the query
    cursor.execute(sql)
    conn.commit()
    
    print("Table computer_status updated successfully to include 'maintenance' status.")
    
except mysql.connector.Error as err:
    print(f"Error updating table: {err}")
    
finally:
    # Close connection
    if 'cursor' in locals():
        cursor.close()
    if 'conn' in locals():
        conn.close() 

File: update_db.php
<?php
// Database connection parameters
$servername = "localhost";
$username = "root";
$password = "";
$dbname = "sysarch";

// Create connection
$conn = new mysqli($servername, $username, $password, $dbname);

// Check connection
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

echo "Connected to database successfully.<br>";

// SQL query to alter the computer_status table
$sql = "ALTER TABLE computer_status 
        MODIFY COLUMN status ENUM('vacant', 'occupied', 'in_class', 'maintenance') NOT NULL";

// Execute the query
if ($conn->query($sql) === TRUE) {
    echo "Table computer_status updated successfully to include 'maintenance' status.";
} else {
    echo "Error updating table: " . $conn->error;
}

// Close connection
$conn->close();
?> 